src/Patch/Document.js

```diff
@@ -0,0 +1,67 @@
+import Native from './Native';
+import CustomElementInternals from '../CustomElementInternals';
+import * as Utilities from '../Utilities';
+
+import PatchParentNode from './Interface/ParentNode';
+
+/**
+ * @param {!CustomElementInternals} internals
+ */
+export default function(internals) {
+  Utilities.setPropertyUnchecked(Document.prototype, 'createElement',
```

Comment:
As discussed, these three methods need to check for `this === window.document` before customizing.